### PR TITLE
automation scripts added!

### PR DESCRIPTION
**Pull request by nghiant2710**

### CODE REVIEW

File: automation/jenkins_build.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+VERSION=$(<VERSION)
+
+docker build -t docker-test:$VERSION .
+
+docker run --rm --privileged -e BUILDFLAGS -e DOCKER_CLIENTONLY -e DOCKER_EXECDRIVER -e DOCKER_GRAPHDRIVER -e TESTDIRS -e TESTFLAGS -e TIMEOUT -v "`pwd`:/go/src/github.com/docker/docker" "docker-test:$VERSION" hack/make.sh binary cross
+
+## ARM Build
+mkdir -p docker-arm-$VERSION
+rm -rf docker-arm-$VERSION.tar
```
Comment (petrosagg) [inline automation/jenkins_build.sh]: no need for `-r` since you're deleting a file

File: automation/jenkins_build.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+rm -rf docker-arm*.tar.xz
```
Comment (petrosagg) [inline automation/jenkins_build.sh]: same here

File: automation/jenkins_build.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+VERSION=$(<VERSION)
+
+docker build -t docker-test:$VERSION .
+
+docker run --rm --privileged -e BUILDFLAGS -e DOCKER_CLIENTONLY -e DOCKER_EXECDRIVER -e DOCKER_GRAPHDRIVER -e TESTDIRS -e TESTFLAGS -e TIMEOUT -v "`pwd`:/go/src/github.com/docker/docker" "docker-test:$VERSION" hack/make.sh binary cross
```
Comment (petrosagg) [inline automation/jenkins_build.sh]: What are the values of these env vars?

File: automation/jenkins_build.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+cp bundles/$VERSION/cross/linux/arm/docker-$VERSION docker-arm-$VERSION/docker
+tar -cf docker-arm-$VERSION.tar docker-arm-$VERSION && xz docker-arm-$VERSION.tar && rm -rf docker-arm-$VERSION
+
+## i386 Build
+mkdir -p docker-386-$VERSION
+rm -rf docker-386-$VERSION.tar
+rm -rf docker-386*.tar.xz
+
+cp bundles/$VERSION/cross/linux/386/docker-$VERSION docker-386-$VERSION/docker
+tar -cf docker-386-$VERSION.tar docker-386-$VERSION && xz docker-386-$VERSION.tar && rm -rf docker-386-$VERSION
```
Comment (petrosagg) [inline automation/jenkins_build.sh]: `tar -cJf` will create the .tar.xz file directly

File: automation/upload_binary.sh [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+# ENV VARS
+VERSION=$(<VERSION)
+#ACCOUNT=
+#REPO=
+#ACCESS_TOKEN=
+#sourceBranch=
+
+# Create a release
+rm -f request.json response.json
+echo "{
```
Comment (petrosagg) [inline automation/upload_binary.sh]: You can use HEREDOC here like so:

``` bash
cat > request.json <<-EOF
{
    "tag_name": "$VERSION",
    "target_commitish": "$sourceBranch",
    "name": "$VERSION",
    "body": "Release of version $VERSION",
    "draft": false,
    "prerelease": false
}
EOF
```

File: automation/upload_binary.sh [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  \"tag_name\": \"$VERSION\",
+  \"target_commitish\": \"$sourceBranch\",
+  \"name\": \"$VERSION\",
+  \"body\": \"Release of version $VERSION\",
+  \"draft\": false,
+  \"prerelease\": false
+}" > request.json
+curl -d @request.json https://api.github.com/repos/$ACCOUNT/$REPO/releases?access_token=$ACCESS_TOKEN -o response.json
```
Comment (petrosagg) [inline automation/upload_binary.sh:27]: `$ACCOUNT`, `$REPO` `$ACCESS_TOKEN` are commented out?

File: automation/upload_binary.sh [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+# Parse response
+RELEASE_ID=$(sed -n '/id/{p;q;}' response.json)
```
Comment (petrosagg) [inline automation/upload_binary.sh]: Use `jq` to get values from JSON. https://stedolan.github.io/jq/

File: automation/upload_binary.sh [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+RELEASE_ID=${RELEASE_ID:8:-1}
+echo "RELEASE_ID=$RELEASE_ID"
+# Upload data
+# arm
+curl -H "Authorization:token $ACCESS_TOKEN" -H "Content-Type:application/gzip" --data @docker-arm-$VERSION.tar.xz https://uploads.github.com/repos/$ACCOUNT/$REPO/releases/$RELEASE_ID/assets?name=docker-arm-$VERSION.tar.xz
```
Comment (petrosagg) [inline automation/upload_binary.sh]: Space missing after the colon. Also, it's a good idea to put double quotes around all the arguments that have variables in them. For example: `"@docker-arm-$VERSION.tar.xz"`

File: automation/jenkins_build.sh [hunk 9]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+VERSION=$(<VERSION)
+
+docker build -t docker-test:$VERSION .
```
Comment (petrosagg) [inline automation/jenkins_build.sh]: `docker-test`?

File: automation/jenkins_build.sh (hunk 1 above)
Comment (nghiant2710) [inline automation/jenkins_build.sh]: Updated!

File: automation/jenkins_build.sh (hunk 3 above)
Comment (nghiant2710) [inline automation/jenkins_build.sh]: I took this line from rce repo. Removed these env vars!

File: automation/jenkins_build.sh (hunk 4 above)
Comment (nghiant2710) [inline automation/jenkins_build.sh]: Updated!

File: automation/upload_binary.sh (hunk 6 above)
Comment (nghiant2710) [inline automation/upload_binary.sh:27]: keep these vars here to remind which env vars are available

File: automation/jenkins_build.sh (hunk 9 above)
Comment (nghiant2710) [inline automation/jenkins_build.sh]: I changed it to docker-builder

File: automation/upload_binary.sh [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+}
+EOF
+curl -d @request.json https://api.github.com/repos/$ACCOUNT/$REPO/releases?access_token=$ACCESS_TOKEN -o response.json
+# Parse response
+RELEASE_ID=$(cat response.json | jq '.id')
+echo "RELEASE_ID=$RELEASE_ID"
+
+# Upload data
+# arm
+curl -H "Authorization:token $ACCESS_TOKEN" -H "Content-Type:application/gzip" --data "@docker-arm-$VERSION.tar.xz" https://uploads.github.com/repos/$ACCOUNT/$REPO/releases/$RELEASE_ID/assets?name=docker-arm-$VERSION.tar.xz
```
Comment (petrosagg) [inline automation/upload_binary.sh]: The content type isn't gzip

File: automation/upload_binary.sh [hunk 11]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+# ENV VARS
+VERSION=$(<VERSION)
+#ACCOUNT=
```
Comment (petrosagg) [inline automation/upload_binary.sh:5]: The script should print a message and exit if these are unset

Comment (nghiant2710) [inline automation/upload_binary.sh:5]: @petrosagg , updated!

**Pull request merged**